vignettes/gsmExtensions.Rmd

```diff
@@ -0,0 +1,422 @@
+      strGroupLevel: GroupLevel
+      strNumeratorMethod: Count
+      strDenominatorMethod: Sum
+      strDenominatorCol: timeonstudy
+  - name: Transform_Rate
+    output: Analysis_Transformed
+    params:
+      dfInput: Analysis_Input
+  - name: Analyze_NormalApprox
+    output: Analysis_Analyzed
+    params:
+      dfTransformed: Analysis_Transformed
+      strType: AnalysisType
+  - name: Flag_NormalApprox
+    output: Analysis_Flagged
+    params:
+      dfAnalyzed: Analysis_Analyzed
+      vThreshold: vThreshold
+  - name: Summarize
+    output: Analysis_Summary
+    params:
+      dfFlagged: Analysis_Flagged
+      nMinDenominator: nMinDenominator
+  - name: list
+    output: kri0001
+    params:
+      id: ID
+      input: Analysis_Input
+      transformed: Analysis_Transformed
+      analyzed: Analysis_Analyzed
+      flagged: Analysis_Flagged
+      summary: Analysis_Summary
+```
+
+#### `Report` steps example
+
+In this example, the steps to produce a site-level KRI report is displayed. Here, the only inputs are the `Reporting_*` data, which goes through a simple filtering process via `RunQuery` before the Charts and Report are created in the following two functions
+
+```
+steps:
+  - name: RunQuery
+    output: Reporting_Results_Site
+    params:
+      df: Reporting_Results
+      strQuery: "SELECT * FROM df WHERE GroupLevel == 'Site'"
+  - name: RunQuery
+    output: Reporting_Metrics_Site
+    params:
+      df: Reporting_Metrics
+      strQuery: "SELECT * FROM df WHERE GroupLevel == 'Site'"
+  - name: MakeCharts
+    output: lCharts_Site
+    params:
+      dfResults: Reporting_Results_Site
+      dfGroups: Reporting_Groups
+      dfBounds: Reporting_Bounds
+      dfMetrics: Reporting_Metrics_Site
+  - name: Report_KRI
+    output: lReport 
+    params:
+      lCharts: lCharts_Site
+      dfResults: Reporting_Results_Site
+      dfGroups: Reporting_Groups
+      dfMetrics: Reporting_Metrics_Site
+```
+
+# Directory Structure for Workflows 
+
+Each extension that produces report(s) will have a `workflow` directory in the `inst` of the package that follows a standard structure. This directory will contain 4 folders in which to store the `yaml` workflow files that map data, perform analysis, produce reporting data, and generate the output of a module. Each module output requires it's own unique `yaml` in the `4_modules` folder, which will take inputs generated from the previous three directories.
+
+
+### `/1_mappings`
+
+The mappings folder contains all of the mappings from `Raw_*` data to `Mapped_*` data. Each file within this directory is to be named for the data table it is creating, minus the `Mapped_` suffix. The `yaml`s will contain the three required sections, which are discussed in detail in the `Module Configuration` section above. The `yaml`s in this folder will be combined via `CombineSpecs()` to create a master spec that defines all necessary tables and columns for the module(s) in this package.
```

Comment:
```suggestion
The mappings folder contains all of the mappings from `Raw_*` data to `Mapped_*` data. Each file within this directory is to be named for the data table it is creating, minus the `Mapped_` suffix. The `yaml`'s will contain the three required sections, which are discussed in detail in the `Module Configuration` section above. The `yaml`'s in this folder will be combined via `CombineSpecs()` to create a master spec that defines all necessary tables and columns for the module(s) in this package.
```

`yaml`s?
`yamls`?
`yaml`'s
`yaml`-s

Idk which looks best?

Reply:
yeah, this is something i've been thinking about as i was writing this up. i think either `yamls` or just not adding ticks at all and just typing it all in plain text make the most sense to me.